Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+                
+                if($FrontDoorEndpoint)
+                {
+                Write-Host "Front Door Configurations successfully fetched." -ForegroundColor $([Constants]::MessageType.Update)
+                Write-Host $([Constants]::SingleDashLine)
+                }
+
+                if($null -ne $FrontDoorEndpoint)
+                {
+                foreach ($item in $FrontDoorEndpoint) 
+                {
+                    $domainName = $item.Name
+                    $domainResourceGroupName = $item.ResourceGroupName
+                    $TypeOfCertificate= @()
+                    $MinTLSVersion = @()
+                    
+                    if($null -ne $ClassicAccessToken)
+                    {
+                        $header = "Bearer " + $ClassicAccessToken
+                        $headers = @{"Authorization"=$header;"Content-Type"="application/json"; "x-ms-version" ="2013-08-01"}
+                        $uri = [string]:: Format("{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.Cdn/profiles/{3}/customDomains/{4}?api-version=2021-06-01",$ResourceAppIdURI,$SubscriptionId,$domainResourceGroupName,$resourceName,$domainName)
+                        $customDomainResponse = Invoke-WebRequest -Method GET -Uri $uri -Headers $headers -UseBasicParsing
+                    
+                        if($customDomainResponse.StatusCode -ge 200 -and $customDomainResponse.StatusCode -le 399)
+                        {
+                            if($null -ne $customDomainResponse.Content)
+                            {
+                                $custom = $customDomainResponse.Content | ConvertFrom-Json  
+                                if($null -ne $custom)
+                                {
+                                    $TypeOfCertificate = $custom.properties.tlsSettings.certificateType
+                                    $MinTLSVersion = $custom.properties.tlsSettings.minimumTlsVersion
+                                }
+                            }
+                        }        
+                    }
+                    if($null -ne $MinTLSVersion)
+                    {
+                        if($MinTLSVersion -ne $requiredMinTLSVersion)
+                        {
+                            $frontDoorEp.DomainName = $domainName
+                            $frontDoorEp.FrontDoorName = $resourceName
+                            $frontDoorEp.MinTLSVersion = $MinTLSVersion
+                            $frontDoorEp.ResourceGroup = $resourceGroupName
+                            $frontDoorEp.TypeOfCertificate = $TypeOfCertificate
+                            $frontDoorEp.isMinTLSVersionSetOnCustomDomain = $false
+                        }
+                        else
+                        {
+                            $frontDoorsEndpointWithtReqMinTLSVersion = $item
+                        }
+                    }
+                    if($null -ne $frontDoorEp)
```

Comment:
If we have multiple custom domain in profile and very first is non-compliant then this condition will be true for all next coming endpoint and all will be marked as non-compliant even if they are compliant. Plz check, if my doubt is correct.

Reply:
Fixed issue with multiple domains.